### PR TITLE
Update argocd image to v2.4.0-rc1

### PR DESCRIPTION
**Pull request by chetan-rns**
**What type of PR is this?**
> Uncomment only one ` /kind` line, and delete the rest.
> For example, `> /kind bug` would simply become: `/kind bug`

/kind chore

**What does this PR do / why we need it**:
* Update argocd image to v2.4.0-rc1
* Remove applicationset image since applicationset is bundled in the argocd image 
* Update CRDs

**Have you updated the necessary documentation?**

* [ ] Documentation update is required by this PR.
* [ ] Documentation has been updated.

**Which issue(s) this PR fixes**:

Fixes #?

**How to test changes / Special notes to the reviewer**:

### CODE REVIEW

**Comment by chetan-rns** [comment]
@iam-veeramalla Any idea why the e2e test job is failing? It only says `Error: Process completed with exit code 1.`  I did try to re-trigger  the job, but no luck

File: build/util/Dockerfile [hunk 1]
```diff
@@ -1,5 +1,5 @@
-# Argo CD v2.3.3
-FROM quay.io/argoproj/argocd@sha256:dd738f234fcdb0aac8631a0fd1aafbbcd86f936480b06e8377b033ef7a764f71
+# Argo CD v2.4.0-rc1
+FROM quay.io/argoproj/argocd@sha256:d8f845a1263dd6d01ee3073a5ba168d7522648e3
```
Comment (jopit) [inline build/util/Dockerfile:2]: The image tag for the quay.io container image for Argo CD v2.4.0-rc1 is 92ea7377a66fa3e79dd1d7c804307a8b8829049fcf30df3884be764945d8ce7e

**Review by jopit — CHANGES_REQUESTED** [review]

File: build/util/Dockerfile (hunk 1 above)
Comment (chetan-rns) [reply build/util/Dockerfile]: Thanks! I've updated it with the right SHA

**Review by jopit — APPROVED** [review]
LGTM

**Pull request merged**